client/src/components/SelectAvatar.svelte

```diff
@@ -59,13 +64,18 @@
 </script>
 
 <div class="grid {lobby != undefined ? 'lobby' : ''}">
-  {#each avatarChoices as { avatar, altText, displayName, available, selected }}
-    <button class="avatar" on:click={() => selectAvatar(avatar)} disabled={!available} aria-selected={selected}>
-      <img src="/avatars/{avatar}.webp" alt={altText} />
-      {#if lobby != undefined}
-        <span class="mdc-typography--subtitle1">{displayName ?? ""}</span>
+  {#each avatarChoices as { avatar, altText, displayName, uid, available, selected }}
+    <div class="parent">
+      {#if !available && $user !== null && lobby?.uids[0] == $user.uid}
```

Comment:
you could also check to make sure the lobbyCode isn't undefined so you could just pass in a string into your player menu component.

The lobby component passses in lobby code which is type string not (string | undefined) into select avatar. Nate made it undefined | string because this is being used else where.